components/google_calendar/sources/upcoming-event-alert/upcoming-event-alert.mjs

```diff
@@ -0,0 +1,112 @@
+import taskScheduler from "../../../pipedream/sources/new-scheduled-tasks/new-scheduled-tasks.mjs";
+import googleCalendar from "../../google_calendar.app.mjs";
+import { axios } from "@pipedream/platform";
+
+const docLink = "https://pipedream.com/docs/examples/waiting-to-execute-next-step-of-workflow/#step-1-create-a-task-scheduler-event-source";
+
+export default {
+  key: "google_calendar-upcoming-event-alert",
+  name: "Upcoming Event Alert",
+  description: `Triggers based on a time interval before an upcoming event in the calendar.
+    This source uses Pipedream's Task Scheduler.
+    [See here](${docLink}) for more info and instructions to connect your Pipedream account.`,
+  version: "0.0.1",
+  type: "source",
+  props: {
+    pipedream: taskScheduler.props.pipedream,
+    googleCalendar,
+    db: "$.service.db",
+    http: "$.interface.http",
+    calendarId: {
+      propDefinition: [
+        googleCalendar,
+        "calendarId",
+      ],
+    },
+    eventId: {
+      propDefinition: [
+        googleCalendar,
+        "eventId",
+        (c) => ({
+          calendarId: c.calendarId,
+        }),
+      ],
+    },
+    time: {
+      type: "integer",
+      label: "Minutes Before",
+      description: `Number of minutes to trigger before the calendar event.
+        \\
+        You may use a custom expression to express in another unit. E.g. \`{{ 60 * 2 }}\` is 2 hours.`,
```

Comment:
@andrewjschuang is there a specific use case you see for entering a custom expression? Since it's the trigger, obviously users won't be referencing previous steps. I can't think of a reason users would not just enter absolute numbers, can you?

Reply:
Just so that it is easier to put in another time unit, like hours or day